pandas/core/dtypes/common.py

```diff
@@ -692,6 +692,21 @@ def is_dtype_equal(source, target):
         return False
 
 
+def _is_dtype_union_equal(source, target):
+    """
+    Check whether two arrays have compatible dtypes to do a unoin.
+    numpy types are checked with ``is_dtype_equal``. Extension types are
+    checked separately.
```

Comment:
can you add Parameters section

this is a private module so could do: ``is_dtype_union_equal``